pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -450,8 +451,13 @@ class ModelInterpreter implements Serializable {
     def withEnvBlock(Map<String,Closure> envVars, Closure body) {
         if (envVars != null && !envVars.isEmpty()) {
             List<String> evaledEnv = envVars.collect { k, v ->
-                "${k}=${v.call()}"
-            }
+                try{
+                    "${k}=${v.call()}"
+                }catch (NullPointerException e){
+
+                    throw new IllegalArgumentException(format("Invalid var declared in environment: %s", k))
```

Comment:
Maybe reword this to be more clear - is it that the environment variable `FOO` is invalid, or that something went wrong with its value?

Reply:
Changed to: 

> Invalid variable (FOO) declared in environment; There are some problems with its value